dotcom-rendering/src/server/lib/get-content-from-url.js

```diff
@@ -52,27 +46,32 @@ exports.default = getContentFromURL;
 
 /**
  * @param {string} requestUrl
- * @returns {string}
+ * @returns {URL | undefined} the parsed URL, or false if there’s no fully qualified path
  */
 const parseURL = (requestUrl) => {
 	const url = decodeURIComponent(requestUrl.split('/').slice(2).join('/'));
 
-	return requestUrl.startsWith('/AMP') ? url.replace('www', 'amp') : url;
+	try {
+		return new URL(
+			requestUrl.startsWith('/AMP') ? url.replace('www', 'amp') : url,
```

Comment:
Rather than replacing the URL here, would it be better to redirect the user? Seems like it would provide a clearer experience

Reply:
Oh, I like that, that’s a better idea!

Reply:
https://user-images.githubusercontent.com/76776/220674343-1bd1c72c-6f90-4570-a36b-0349d1912bf6.mov